bot/common/tasks/tasks.py

```diff
@@ -1,25 +1,56 @@
-import threading
 import logging
 import asyncio
+import discord
+import pytz
 
 from abc import ABC, abstractmethod
 from datetime import timedelta, datetime, time
+from discord.ext import tasks, commands
 
 from bot.common.tasks.weekly_contributions import send_weekly_contribution_reports
-from bot.common.cache import Cache, RedisCache
+from bot.common.cache import Cache
+from bot.constants import BotTasks as TaskConstants
 
 logger = logging.getLogger(__name__)
 
-TASK_LOOP_INTERVAL_MINUTES = 10
-REPORT_LAST_SENT_DATETIME_CACHE_KEY = "contribution_report_last_sent"
+# TODO: tz?
 DATETIME_CACHE_FMT = "%m/%d/%Y, %H:%M:%S"
 
 
 class Cadence(ABC):
-    # Returns the number of seconds until the task can be run. Negative
-    # values reflect that the task is due to fire.
+    """
+    Abstract base class for cadences. Objects that implement this ABC
+    are intended to help with repetive tasks that fire on a specific
+    schedule.
+    """
+
+    """
+    Returns the number of seconds until the task can be run. Negative
+    values reflect that the task is due to fire. When negative values
+    are returned are at the discretion of the implementing class.
+    """
```

Comment:
Best to combine these in a single doc string. I am not sure but it could omit one of the strings in a __doc__ call on the function.